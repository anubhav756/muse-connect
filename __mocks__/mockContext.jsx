import React from 'react';
import PropTypes from 'prop-types'
import { getMuiTheme } from 'material-ui/styles'

export function mockStore(dummyStore = {}, dispatch = {}) {
  return {
    getState: () => dummyStore,
    dispatch: () => dispatch,
    subscribe: () => {}
  }
}

export default function (dummyStore, dispatch) {
  return {
    context: {
      store: mockStore(dummyStore, dispatch),
      muiTheme: getMuiTheme(),
    },
    childContextTypes: {
      muiTheme: PropTypes.object,
      store: PropTypes.object
    }
  }
}
