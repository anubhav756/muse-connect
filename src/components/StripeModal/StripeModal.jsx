import React, { Component } from 'react';
import PropTypes from 'prop-types'
import { connect } from 'react-redux';
import StripeCheckout from 'react-stripe-checkout';
import {
  saveToken as _saveToken,
  usePromo as _usePromo
} from '../../redux/modules/subscription';
import { capitalize } from '../../libs/helpers/common';

class StripeModal extends Component {
  constructor() {
    super();

    this.state = {
      tokenReceived: false,
      forceOpenModal: false
    };

    this.onToken = this.onToken.bind(this);
    this.onClose = this.onClose.bind(this);
    this.getCheckoutURL = this.getCheckoutURL.bind(this);
  }
  onToken({ id: token }) {
    const {
      planID,
      user: { email },
      saveToken,
      promoID,
      purID,
      callback
    } = this.props;

    this.setState({ tokenReceived: true });

    saveToken({
      email,
      token,
      planID,
      ...(promoID ? { promoID } : null),
      ...(purID ? { purID } : null)
    },
      callback
    );
  }
  onClose() {
    const { callback } = this.props;
    const { tokenReceived } = this.state;

    if (!tokenReceived)
      callback(true);

    this.setState({ tokenReceived: false, forceOpenModal: false });
  }
  getCheckoutURL() {
    const { promoID, usePromo, callback } = this.props;

    usePromo(promoID, (err, checkoutURL) => {
      if (err) {
        return callback(err);
      }

      if (checkoutURL) {
        window.open(checkoutURL, '_self');
      } else {
        this.setState({ forceOpenModal: true }, () => this.setState({ forceOpenModal: false }));
      }
    });
  }
  render() {
    const {
      title,
      button: _button,
      disabled,
      amount,
      trialDays,
      currency,
      user: { email },
      stripeKey,
      promoID,
      frequency
    } = this.props;
    const { forceOpenModal } = this.state;
    const button = React.cloneElement(_button, { disabled });

    if (!forceOpenModal) {
      if (disabled)
        return button;

      if (promoID)
        return <div onClick={this.getCheckoutURL}>{button}</div>;
    }

    let panelLabel = trialDays ? '$0.00 - Start Free Trial' : frequency || 'Pay';
    panelLabel = capitalize(panelLabel);

    return (
      <StripeCheckout
        name="Muse"
        desktopShowModal={forceOpenModal}
        description={title}
        image="https://www.choosemuse.com/wp-content/uploads/2014/10/muse_logo_noTag1.png"
        panelLabel={panelLabel}
        amount={trialDays ? null : parseFloat(amount) * 100}
        currency={currency}
        stripeKey={stripeKey}
        email={email}
        token={this.onToken}
        triggerEvent="onClick"
        opened={this.onOpen}
        closed={this.onClose}
      >
        {button}
      </StripeCheckout>
    );
  }
}
function enabledOnlyProp(props, propName) {
  if (!props.disabled && !props[propName])
    return new Error(`"${propName}" required when "disabled" is set to true`);
}

StripeModal.propTypes = {
  planID: PropTypes.string,
  title: PropTypes.string,
  button: PropTypes.object.isRequired,
  disabled: PropTypes.bool,
  frequency: PropTypes.string,
  trialDays: PropTypes.number,
  amount: enabledOnlyProp,
  currency: enabledOnlyProp,
  stripeKey: enabledOnlyProp,
  user: PropTypes.shape({
    email: PropTypes.string.isRequired
  }).isRequired,
  saveToken: PropTypes.func.isRequired,
  usePromo: PropTypes.func.isRequired,
  callback: PropTypes.func.isRequired,
  promoID: PropTypes.string,
  purID: PropTypes.string
}
StripeModal.defaultProps = {
  title: 'Muse',
  trialDays: 0,
  disabled: false,
  checkoutURL: '',
  frequency: null,
  amount: 0,
  currency: 'USD',
  stripeKey: null,
  planID: null,
  promoID: null,
  purID: null
}

export default connect(
  null,
  ({
    saveToken: _saveToken,
    usePromo: _usePromo
  })
)(StripeModal);
