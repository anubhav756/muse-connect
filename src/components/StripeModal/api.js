import apiCall from '../../libs/api/apiRequest';
import endPoints from '../../libs/api/endpoints';

export default function saveToken(payload) {
  const endpoint = endPoints.subscriptions;
  const method = 'put';
  return apiCall({ endpoint, method, payload });
}

export function usePromo(promoID) {
  const endpoint = `${endPoints.promos}/${promoID}`;
  const method = 'post';

  return apiCall({ endpoint, method });
}
