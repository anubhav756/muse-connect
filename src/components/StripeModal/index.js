import StripeModal from './StripeModal';

export default StripeModal;
