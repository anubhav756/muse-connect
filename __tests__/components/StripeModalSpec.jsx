import React from 'react';
import { mount } from 'enzyme'
import toJSON from 'enzyme-to-json';
import sinon from 'sinon';
import StripeModal from '../../src/components/StripeModal';
import mockContext from '../../__mocks__/mockContext';

jest.mock('../../src/redux/modules/subscription');


describe('<StripeModal />', () => {
  const buildWrapper = (props, store) => (
    mount(<StripeModal {...props} />, mockContext(store))
  )

  it('renders correctly', () => {
    const component = buildWrapper({
      saveToken: () => {},
      button: <div />,
      amount: 100,
      stripeKey: 'test_stripeKey',
      user: { email: 'test@email.com' },
      callback: () => {}
    }, {
      windowDimension: { innerWidth: 1200 }
    });
    expect(toJSON(component)).toMatchSnapshot();
  });

  it('renders correctly with promoID', () => {
    const component = buildWrapper({
      saveToken: () => {},
      button: <div />,
      amount: 100,
      stripeKey: 'test_stripeKey',
      user: { email: 'test@email.com' },
      callback: () => {},
      promoID: 'test_promoID'
    }, {
      windowDimension: { innerWidth: 1200 }
    });
    expect(toJSON(component)).toMatchSnapshot();
  });

  it('gets checkoutURL with promoID correctly', () => {
    const component = buildWrapper({
      saveToken: () => {},
      button: <div />,
      amount: 100,
      stripeKey: 'test_stripeKey',
      user: { email: 'test@email.com' },
      callback: () => {},
      promoID: 'test_promoID'
    }, {
      windowDimension: { innerWidth: 1200 }
    });
    expect(toJSON(component)).toMatchSnapshot();
    component.find('StripeModal').instance().getCheckoutURL();
    expect(toJSON(component)).toMatchSnapshot();
  });

  it('renders correctly when disabled', () => {
    const component = buildWrapper({
      saveToken: () => {},
      button: <div />,
      amount: 100,
      stripeKey: 'test_stripeKey',
      user: { email: 'test@email.com' },
      callback: () => {},
      disabled: true
    }, {
      windowDimension: { innerWidth: 1200 }
    });
    expect(toJSON(component)).toMatchSnapshot();
  });

  it('enabled only prop checked correctly', () => {
    const component = buildWrapper({
      saveToken: () => {},
      button: <div />,
      stripeKey: 'test_stripeKey',
      user: { email: 'test@email.com' },
      callback: () => {}
    }, {
      windowDimension: { innerWidth: 1200 }
    });
    expect(toJSON(component)).toMatchSnapshot();
  });

  it('stripeCheckout onClose method closes correctly', () => {
    const component = buildWrapper({
      saveToken: () => {},
      button: <div />,
      amount: 100,
      stripeKey: 'test_stripeKey',
      user: { email: 'test@email.com' },
      callback: () => {}
    }, {
      windowDimension: { innerWidth: 1200 }
    });
    expect(toJSON(component)).toMatchSnapshot();
    component.find('ReactStripeCheckout').props().closed();
    expect(toJSON(component)).toMatchSnapshot();
  });

  it('stripeCheckout onClose method without token invokes callback with true', () => {
    const spyCallback = sinon.spy();
    const component = buildWrapper({
      saveToken: () => {},
      button: <div />,
      amount: 100,
      stripeKey: 'test_stripeKey',
      user: { email: 'test@email.com' },
      callback: spyCallback
    }, {
      windowDimension: { innerWidth: 1200 }
    });
    expect(toJSON(component)).toMatchSnapshot();
    component.find('ReactStripeCheckout').props().closed();
    expect(toJSON(component)).toMatchSnapshot();
    expect(spyCallback.calledWith()).toEqual(true);
  });

  it('stripeCheckout onToken method saves token correctly', () => {
    const component = buildWrapper({
      saveToken: () => {},
      button: <div />,
      amount: 100,
      stripeKey: 'test_stripeKey',
      user: { email: 'test@email.com' },
      callback: () => {}
    }, {
      windowDimension: { innerWidth: 1200 }
    });
    expect(toJSON(component)).toMatchSnapshot();
    component.find('ReactStripeCheckout').props().token({ id: 'test_token' });
    expect(toJSON(component)).toMatchSnapshot();
  });

  it('stripeCheckout onClose after onToken invokes callback correctly', () => {
    const component = buildWrapper({
      saveToken: () => {},
      button: <div />,
      amount: 100,
      stripeKey: 'test_stripeKey',
      user: { email: 'test@email.com' },
      callback: () => {}
    }, {
      windowDimension: { innerWidth: 1200 }
    });
    expect(toJSON(component)).toMatchSnapshot();
    component.find('ReactStripeCheckout').props().token({ id: 'test_token' });
    expect(toJSON(component)).toMatchSnapshot();
    component.find('ReactStripeCheckout').props().closed();
    expect(toJSON(component)).toMatchSnapshot();
  });
});
