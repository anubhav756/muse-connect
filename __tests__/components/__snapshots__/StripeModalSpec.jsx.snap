// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StripeModal /> enabled only prop checked correctly 1`] = `
<Connect(StripeModal)
  button={<div />}
  callback={[Function]}
  saveToken={[Function]}
  stripeKey="test_stripeKey"
  user={
    Object {
      "email": "test@email.com",
    }
  }
>
  <StripeModal
    amount={0}
    button={<div />}
    callback={[Function]}
    checkoutURL=""
    currency="USD"
    disabled={false}
    frequency={null}
    planID={null}
    promoID={null}
    purID={null}
    saveToken={[Function]}
    stripeKey="test_stripeKey"
    title="Muse"
    trialDays={0}
    usePromo={[Function]}
    user={
      Object {
        "email": "test@email.com",
      }
    }
  >
    <ReactStripeCheckout
      ComponentClass="span"
      amount={0}
      className="StripeCheckout"
      closed={[Function]}
      currency="USD"
      description="Muse"
      desktopShowModal={false}
      email="test@email.com"
      image="https://www.choosemuse.com/wp-content/uploads/2014/10/muse_logo_noTag1.png"
      label="Pay With Card"
      locale="auto"
      name="Muse"
      opened={undefined}
      panelLabel="Pay"
      reconfigureOnUpdate={false}
      stripeKey="test_stripeKey"
      token={[Function]}
      triggerEvent="onClick"
    >
      <span
        onClick={[Function]}
      >
        <div
          disabled={false}
        />
      </span>
    </ReactStripeCheckout>
  </StripeModal>
</Connect(StripeModal)>
`;

exports[`<StripeModal /> gets checkoutURL with promoID correctly 1`] = `
<Connect(StripeModal)
  amount={100}
  button={<div />}
  callback={[Function]}
  promoID="test_promoID"
  saveToken={[Function]}
  stripeKey="test_stripeKey"
  user={
    Object {
      "email": "test@email.com",
    }
  }
>
  <StripeModal
    amount={100}
    button={<div />}
    callback={[Function]}
    checkoutURL=""
    currency="USD"
    disabled={false}
    frequency={null}
    planID={null}
    promoID="test_promoID"
    purID={null}
    saveToken={[Function]}
    stripeKey="test_stripeKey"
    title="Muse"
    trialDays={0}
    usePromo={[Function]}
    user={
      Object {
        "email": "test@email.com",
      }
    }
  >
    <div
      onClick={[Function]}
    >
      <div
        disabled={false}
      />
    </div>
  </StripeModal>
</Connect(StripeModal)>
`;

exports[`<StripeModal /> gets checkoutURL with promoID correctly 2`] = `
<Connect(StripeModal)
  amount={100}
  button={<div />}
  callback={[Function]}
  promoID="test_promoID"
  saveToken={[Function]}
  stripeKey="test_stripeKey"
  user={
    Object {
      "email": "test@email.com",
    }
  }
>
  <StripeModal
    amount={100}
    button={<div />}
    callback={[Function]}
    checkoutURL=""
    currency="USD"
    disabled={false}
    frequency={null}
    planID={null}
    promoID="test_promoID"
    purID={null}
    saveToken={[Function]}
    stripeKey="test_stripeKey"
    title="Muse"
    trialDays={0}
    usePromo={[Function]}
    user={
      Object {
        "email": "test@email.com",
      }
    }
  >
    <div
      onClick={[Function]}
    >
      <div
        disabled={false}
      />
    </div>
  </StripeModal>
</Connect(StripeModal)>
`;

exports[`<StripeModal /> renders correctly 1`] = `
<Connect(StripeModal)
  amount={100}
  button={<div />}
  callback={[Function]}
  saveToken={[Function]}
  stripeKey="test_stripeKey"
  user={
    Object {
      "email": "test@email.com",
    }
  }
>
  <StripeModal
    amount={100}
    button={<div />}
    callback={[Function]}
    checkoutURL=""
    currency="USD"
    disabled={false}
    frequency={null}
    planID={null}
    promoID={null}
    purID={null}
    saveToken={[Function]}
    stripeKey="test_stripeKey"
    title="Muse"
    trialDays={0}
    usePromo={[Function]}
    user={
      Object {
        "email": "test@email.com",
      }
    }
  >
    <ReactStripeCheckout
      ComponentClass="span"
      amount={10000}
      className="StripeCheckout"
      closed={[Function]}
      currency="USD"
      description="Muse"
      desktopShowModal={false}
      email="test@email.com"
      image="https://www.choosemuse.com/wp-content/uploads/2014/10/muse_logo_noTag1.png"
      label="Pay With Card"
      locale="auto"
      name="Muse"
      opened={undefined}
      panelLabel="Pay"
      reconfigureOnUpdate={false}
      stripeKey="test_stripeKey"
      token={[Function]}
      triggerEvent="onClick"
    >
      <span
        onClick={[Function]}
      >
        <div
          disabled={false}
        />
      </span>
    </ReactStripeCheckout>
  </StripeModal>
</Connect(StripeModal)>
`;

exports[`<StripeModal /> renders correctly when disabled 1`] = `
<Connect(StripeModal)
  amount={100}
  button={<div />}
  callback={[Function]}
  disabled={true}
  saveToken={[Function]}
  stripeKey="test_stripeKey"
  user={
    Object {
      "email": "test@email.com",
    }
  }
>
  <StripeModal
    amount={100}
    button={<div />}
    callback={[Function]}
    checkoutURL=""
    currency="USD"
    disabled={true}
    frequency={null}
    planID={null}
    promoID={null}
    purID={null}
    saveToken={[Function]}
    stripeKey="test_stripeKey"
    title="Muse"
    trialDays={0}
    usePromo={[Function]}
    user={
      Object {
        "email": "test@email.com",
      }
    }
  >
    <div
      disabled={true}
    />
  </StripeModal>
</Connect(StripeModal)>
`;

exports[`<StripeModal /> renders correctly with promoID 1`] = `
<Connect(StripeModal)
  amount={100}
  button={<div />}
  callback={[Function]}
  promoID="test_promoID"
  saveToken={[Function]}
  stripeKey="test_stripeKey"
  user={
    Object {
      "email": "test@email.com",
    }
  }
>
  <StripeModal
    amount={100}
    button={<div />}
    callback={[Function]}
    checkoutURL=""
    currency="USD"
    disabled={false}
    frequency={null}
    planID={null}
    promoID="test_promoID"
    purID={null}
    saveToken={[Function]}
    stripeKey="test_stripeKey"
    title="Muse"
    trialDays={0}
    usePromo={[Function]}
    user={
      Object {
        "email": "test@email.com",
      }
    }
  >
    <div
      onClick={[Function]}
    >
      <div
        disabled={false}
      />
    </div>
  </StripeModal>
</Connect(StripeModal)>
`;

exports[`<StripeModal /> stripeCheckout onClose after onToken invokes callback correctly 1`] = `
<Connect(StripeModal)
  amount={100}
  button={<div />}
  callback={[Function]}
  saveToken={[Function]}
  stripeKey="test_stripeKey"
  user={
    Object {
      "email": "test@email.com",
    }
  }
>
  <StripeModal
    amount={100}
    button={<div />}
    callback={[Function]}
    checkoutURL=""
    currency="USD"
    disabled={false}
    frequency={null}
    planID={null}
    promoID={null}
    purID={null}
    saveToken={[Function]}
    stripeKey="test_stripeKey"
    title="Muse"
    trialDays={0}
    usePromo={[Function]}
    user={
      Object {
        "email": "test@email.com",
      }
    }
  >
    <ReactStripeCheckout
      ComponentClass="span"
      amount={10000}
      className="StripeCheckout"
      closed={[Function]}
      currency="USD"
      description="Muse"
      desktopShowModal={false}
      email="test@email.com"
      image="https://www.choosemuse.com/wp-content/uploads/2014/10/muse_logo_noTag1.png"
      label="Pay With Card"
      locale="auto"
      name="Muse"
      opened={undefined}
      panelLabel="Pay"
      reconfigureOnUpdate={false}
      stripeKey="test_stripeKey"
      token={[Function]}
      triggerEvent="onClick"
    >
      <span
        onClick={[Function]}
      >
        <div
          disabled={false}
        />
      </span>
    </ReactStripeCheckout>
  </StripeModal>
</Connect(StripeModal)>
`;

exports[`<StripeModal /> stripeCheckout onClose after onToken invokes callback correctly 2`] = `
<Connect(StripeModal)
  amount={100}
  button={<div />}
  callback={[Function]}
  saveToken={[Function]}
  stripeKey="test_stripeKey"
  user={
    Object {
      "email": "test@email.com",
    }
  }
>
  <StripeModal
    amount={100}
    button={<div />}
    callback={[Function]}
    checkoutURL=""
    currency="USD"
    disabled={false}
    frequency={null}
    planID={null}
    promoID={null}
    purID={null}
    saveToken={[Function]}
    stripeKey="test_stripeKey"
    title="Muse"
    trialDays={0}
    usePromo={[Function]}
    user={
      Object {
        "email": "test@email.com",
      }
    }
  >
    <ReactStripeCheckout
      ComponentClass="span"
      amount={10000}
      className="StripeCheckout"
      closed={[Function]}
      currency="USD"
      description="Muse"
      desktopShowModal={false}
      email="test@email.com"
      image="https://www.choosemuse.com/wp-content/uploads/2014/10/muse_logo_noTag1.png"
      label="Pay With Card"
      locale="auto"
      name="Muse"
      opened={undefined}
      panelLabel="Pay"
      reconfigureOnUpdate={false}
      stripeKey="test_stripeKey"
      token={[Function]}
      triggerEvent="onClick"
    >
      <span
        onClick={[Function]}
      >
        <div
          disabled={false}
        />
      </span>
    </ReactStripeCheckout>
  </StripeModal>
</Connect(StripeModal)>
`;

exports[`<StripeModal /> stripeCheckout onClose after onToken invokes callback correctly 3`] = `
<Connect(StripeModal)
  amount={100}
  button={<div />}
  callback={[Function]}
  saveToken={[Function]}
  stripeKey="test_stripeKey"
  user={
    Object {
      "email": "test@email.com",
    }
  }
>
  <StripeModal
    amount={100}
    button={<div />}
    callback={[Function]}
    checkoutURL=""
    currency="USD"
    disabled={false}
    frequency={null}
    planID={null}
    promoID={null}
    purID={null}
    saveToken={[Function]}
    stripeKey="test_stripeKey"
    title="Muse"
    trialDays={0}
    usePromo={[Function]}
    user={
      Object {
        "email": "test@email.com",
      }
    }
  >
    <ReactStripeCheckout
      ComponentClass="span"
      amount={10000}
      className="StripeCheckout"
      closed={[Function]}
      currency="USD"
      description="Muse"
      desktopShowModal={false}
      email="test@email.com"
      image="https://www.choosemuse.com/wp-content/uploads/2014/10/muse_logo_noTag1.png"
      label="Pay With Card"
      locale="auto"
      name="Muse"
      opened={undefined}
      panelLabel="Pay"
      reconfigureOnUpdate={false}
      stripeKey="test_stripeKey"
      token={[Function]}
      triggerEvent="onClick"
    >
      <span
        onClick={[Function]}
      >
        <div
          disabled={false}
        />
      </span>
    </ReactStripeCheckout>
  </StripeModal>
</Connect(StripeModal)>
`;

exports[`<StripeModal /> stripeCheckout onClose method closes correctly 1`] = `
<Connect(StripeModal)
  amount={100}
  button={<div />}
  callback={[Function]}
  saveToken={[Function]}
  stripeKey="test_stripeKey"
  user={
    Object {
      "email": "test@email.com",
    }
  }
>
  <StripeModal
    amount={100}
    button={<div />}
    callback={[Function]}
    checkoutURL=""
    currency="USD"
    disabled={false}
    frequency={null}
    planID={null}
    promoID={null}
    purID={null}
    saveToken={[Function]}
    stripeKey="test_stripeKey"
    title="Muse"
    trialDays={0}
    usePromo={[Function]}
    user={
      Object {
        "email": "test@email.com",
      }
    }
  >
    <ReactStripeCheckout
      ComponentClass="span"
      amount={10000}
      className="StripeCheckout"
      closed={[Function]}
      currency="USD"
      description="Muse"
      desktopShowModal={false}
      email="test@email.com"
      image="https://www.choosemuse.com/wp-content/uploads/2014/10/muse_logo_noTag1.png"
      label="Pay With Card"
      locale="auto"
      name="Muse"
      opened={undefined}
      panelLabel="Pay"
      reconfigureOnUpdate={false}
      stripeKey="test_stripeKey"
      token={[Function]}
      triggerEvent="onClick"
    >
      <span
        onClick={[Function]}
      >
        <div
          disabled={false}
        />
      </span>
    </ReactStripeCheckout>
  </StripeModal>
</Connect(StripeModal)>
`;

exports[`<StripeModal /> stripeCheckout onClose method closes correctly 2`] = `
<Connect(StripeModal)
  amount={100}
  button={<div />}
  callback={[Function]}
  saveToken={[Function]}
  stripeKey="test_stripeKey"
  user={
    Object {
      "email": "test@email.com",
    }
  }
>
  <StripeModal
    amount={100}
    button={<div />}
    callback={[Function]}
    checkoutURL=""
    currency="USD"
    disabled={false}
    frequency={null}
    planID={null}
    promoID={null}
    purID={null}
    saveToken={[Function]}
    stripeKey="test_stripeKey"
    title="Muse"
    trialDays={0}
    usePromo={[Function]}
    user={
      Object {
        "email": "test@email.com",
      }
    }
  >
    <ReactStripeCheckout
      ComponentClass="span"
      amount={10000}
      className="StripeCheckout"
      closed={[Function]}
      currency="USD"
      description="Muse"
      desktopShowModal={false}
      email="test@email.com"
      image="https://www.choosemuse.com/wp-content/uploads/2014/10/muse_logo_noTag1.png"
      label="Pay With Card"
      locale="auto"
      name="Muse"
      opened={undefined}
      panelLabel="Pay"
      reconfigureOnUpdate={false}
      stripeKey="test_stripeKey"
      token={[Function]}
      triggerEvent="onClick"
    >
      <span
        onClick={[Function]}
      >
        <div
          disabled={false}
        />
      </span>
    </ReactStripeCheckout>
  </StripeModal>
</Connect(StripeModal)>
`;

exports[`<StripeModal /> stripeCheckout onClose method without token invokes callback with true 1`] = `
<Connect(StripeModal)
  amount={100}
  button={<div />}
  callback={[Function]}
  saveToken={[Function]}
  stripeKey="test_stripeKey"
  user={
    Object {
      "email": "test@email.com",
    }
  }
>
  <StripeModal
    amount={100}
    button={<div />}
    callback={[Function]}
    checkoutURL=""
    currency="USD"
    disabled={false}
    frequency={null}
    planID={null}
    promoID={null}
    purID={null}
    saveToken={[Function]}
    stripeKey="test_stripeKey"
    title="Muse"
    trialDays={0}
    usePromo={[Function]}
    user={
      Object {
        "email": "test@email.com",
      }
    }
  >
    <ReactStripeCheckout
      ComponentClass="span"
      amount={10000}
      className="StripeCheckout"
      closed={[Function]}
      currency="USD"
      description="Muse"
      desktopShowModal={false}
      email="test@email.com"
      image="https://www.choosemuse.com/wp-content/uploads/2014/10/muse_logo_noTag1.png"
      label="Pay With Card"
      locale="auto"
      name="Muse"
      opened={undefined}
      panelLabel="Pay"
      reconfigureOnUpdate={false}
      stripeKey="test_stripeKey"
      token={[Function]}
      triggerEvent="onClick"
    >
      <span
        onClick={[Function]}
      >
        <div
          disabled={false}
        />
      </span>
    </ReactStripeCheckout>
  </StripeModal>
</Connect(StripeModal)>
`;

exports[`<StripeModal /> stripeCheckout onClose method without token invokes callback with true 2`] = `
<Connect(StripeModal)
  amount={100}
  button={<div />}
  callback={[Function]}
  saveToken={[Function]}
  stripeKey="test_stripeKey"
  user={
    Object {
      "email": "test@email.com",
    }
  }
>
  <StripeModal
    amount={100}
    button={<div />}
    callback={[Function]}
    checkoutURL=""
    currency="USD"
    disabled={false}
    frequency={null}
    planID={null}
    promoID={null}
    purID={null}
    saveToken={[Function]}
    stripeKey="test_stripeKey"
    title="Muse"
    trialDays={0}
    usePromo={[Function]}
    user={
      Object {
        "email": "test@email.com",
      }
    }
  >
    <ReactStripeCheckout
      ComponentClass="span"
      amount={10000}
      className="StripeCheckout"
      closed={[Function]}
      currency="USD"
      description="Muse"
      desktopShowModal={false}
      email="test@email.com"
      image="https://www.choosemuse.com/wp-content/uploads/2014/10/muse_logo_noTag1.png"
      label="Pay With Card"
      locale="auto"
      name="Muse"
      opened={undefined}
      panelLabel="Pay"
      reconfigureOnUpdate={false}
      stripeKey="test_stripeKey"
      token={[Function]}
      triggerEvent="onClick"
    >
      <span
        onClick={[Function]}
      >
        <div
          disabled={false}
        />
      </span>
    </ReactStripeCheckout>
  </StripeModal>
</Connect(StripeModal)>
`;

exports[`<StripeModal /> stripeCheckout onToken method saves token correctly 1`] = `
<Connect(StripeModal)
  amount={100}
  button={<div />}
  callback={[Function]}
  saveToken={[Function]}
  stripeKey="test_stripeKey"
  user={
    Object {
      "email": "test@email.com",
    }
  }
>
  <StripeModal
    amount={100}
    button={<div />}
    callback={[Function]}
    checkoutURL=""
    currency="USD"
    disabled={false}
    frequency={null}
    planID={null}
    promoID={null}
    purID={null}
    saveToken={[Function]}
    stripeKey="test_stripeKey"
    title="Muse"
    trialDays={0}
    usePromo={[Function]}
    user={
      Object {
        "email": "test@email.com",
      }
    }
  >
    <ReactStripeCheckout
      ComponentClass="span"
      amount={10000}
      className="StripeCheckout"
      closed={[Function]}
      currency="USD"
      description="Muse"
      desktopShowModal={false}
      email="test@email.com"
      image="https://www.choosemuse.com/wp-content/uploads/2014/10/muse_logo_noTag1.png"
      label="Pay With Card"
      locale="auto"
      name="Muse"
      opened={undefined}
      panelLabel="Pay"
      reconfigureOnUpdate={false}
      stripeKey="test_stripeKey"
      token={[Function]}
      triggerEvent="onClick"
    >
      <span
        onClick={[Function]}
      >
        <div
          disabled={false}
        />
      </span>
    </ReactStripeCheckout>
  </StripeModal>
</Connect(StripeModal)>
`;

exports[`<StripeModal /> stripeCheckout onToken method saves token correctly 2`] = `
<Connect(StripeModal)
  amount={100}
  button={<div />}
  callback={[Function]}
  saveToken={[Function]}
  stripeKey="test_stripeKey"
  user={
    Object {
      "email": "test@email.com",
    }
  }
>
  <StripeModal
    amount={100}
    button={<div />}
    callback={[Function]}
    checkoutURL=""
    currency="USD"
    disabled={false}
    frequency={null}
    planID={null}
    promoID={null}
    purID={null}
    saveToken={[Function]}
    stripeKey="test_stripeKey"
    title="Muse"
    trialDays={0}
    usePromo={[Function]}
    user={
      Object {
        "email": "test@email.com",
      }
    }
  >
    <ReactStripeCheckout
      ComponentClass="span"
      amount={10000}
      className="StripeCheckout"
      closed={[Function]}
      currency="USD"
      description="Muse"
      desktopShowModal={false}
      email="test@email.com"
      image="https://www.choosemuse.com/wp-content/uploads/2014/10/muse_logo_noTag1.png"
      label="Pay With Card"
      locale="auto"
      name="Muse"
      opened={undefined}
      panelLabel="Pay"
      reconfigureOnUpdate={false}
      stripeKey="test_stripeKey"
      token={[Function]}
      triggerEvent="onClick"
    >
      <span
        onClick={[Function]}
      >
        <div
          disabled={false}
        />
      </span>
    </ReactStripeCheckout>
  </StripeModal>
</Connect(StripeModal)>
`;
